utils/nctl/scripts/config_gen.py

```diff
@@ -0,0 +1,129 @@
+
+    logger.info("Applying overrides to toml file")
+    new_toml_dict = edit_toml(override_dict, toml_dict)
+
+    if args.output_file is not None:
+        logger.info("Generating toml file to: %s", args.output_file)
+        write_toml(args.output_file, new_toml_dict)
+    else:
+        pp.pprint(new_toml_dict)
+
+    logger.info("Finished!")
+
+
+def logger_setup(level):
+    """Sets up logger based on based in log level."""
+    level = level.lower()
+
+    if level == "debug":
+        log_format = (
+            "[%(asctime)s] %(levelname)s [%(name)s.%(funcName)s:%(lineno)d] %(message)s"
+        )
+    else:
+        log_format = "[%(asctime)s] %(levelname)s %(message)s"
+
+    levels = {"info": logging.INFO, "debug": logging.DEBUG, "warning": logging.WARNING}
+    level = levels.get(level)
+
+    logging.basicConfig(
+        stream=sys.stdout, level=level, format=log_format, datefmt="%d-%b-%Y %H:%M:%S"
+    )
+    return logging.getLogger(__name__)
+
+
+def parse_args():
+    """Sets up arguements from CLI input."""
+    parser = argparse.ArgumentParser(
+        description="A script to generate casper-node configs with overridden values"
+    )
+    parser.add_argument("-f", "--toml_file", help="Path to toml file", required=True)
+    parser.add_argument(
+        "-o", "--override_file", help="Path to override file", required=True
+    )
+    parser.add_argument(
+        "-l",
+        "--log_level",
+        help="Sets log level. (default: info)",
+        required=False,
+        default="info",
+        choices=["info", "debug", "warning"],
+    )
+    parser.add_argument(
+        "--output_file",
+        help="Path to write new toml to. If not specified, file is written to stdout",
+        required=False,
+        default=None,
+    )
+    args = parser.parse_args()
+    return args
+
+
+def write_toml(file_path, toml_dict):
+    """Writes TOMLDocument to file path."""
+    logger.debug(pp.pformat(toml_dict))
+    logger.debug("file path: %s", file_path)
+    Path(file_path).write_text(dumps(toml_dict))
+
+
+def edit_toml(override_dict, toml_dict):
+    """Returns updated toml dictionary."""
+    logger.debug(pp.pformat(override_dict))
+    logger.debug(pp.pformat(toml_dict))
+    new_dict = dict(mergedicts(toml_dict, override_dict))
+    return new_dict
+
+
+def mergedicts(dict1, dict2):
+    """Merges two dictionaries where applicable"""
+    for k in set(dict1) | set(dict2):
+        if k in dict1 and k in dict2:
+            if isinstance(dict1[k], dict) and isinstance(dict2[k], dict):
+                yield k, dict(mergedicts(dict1[k], dict2[k]))
+            else:
+                # value from dict2 overrides one in dict1.
+                yield k, dict2[k]
+        elif k in dict1:
+            # not in dict2, keep values of dict1
+            yield k, dict1[k]
+        else:
+            # do nothing if its not present in dict1
+            logger.warning("Skipping %s: not found in original toml", k)
```

Comment:
If it's in `dict2` but not `dict1` it gets skipped? So it's not possible to add entries in the override if they were missing in the original file? Aren't there some options that you can comment out to disable them?

Reply:
added the way to override the behavior: https://github.com/casper-network/casper-node/pull/3159/commits/b8a336822305119edc2c34d89acf8e4614f21fed